proxy/src/telemetry/metrics.rs

```diff
@@ -371,6 +376,17 @@ fn dur_to_ms(dur: Duration) -> u64 {
         })
 }
 
+// ===== impl TransportStats =====
+
+impl From<Protocol> for TransportStats {
```

Comment:
Hm, I can't quite put a finger on it, but something about this impl feels odd to me. It doesn't sound natural that you could go from a `Protocol` into a `TransportStats`...

Reply:
yeah, this is probably not the most semantic way to represent this; i think it came from wanting to be able to pass a function to `or_insert_with` to avoid making a closure, but then, of course, since the function takes a Protocol, it still means we need to make a closure. i'll get rid of it.